### DISCUSSION TITLE
Documentation / Explanation of `_XCTIsTesting`

### DISCUSSION THREAD

**Discussion by dafurman** [pointfreeco/swift-issue-reporting#79] [Q&A] [answered]
I'm curious, what's the benefit of using [_XCTIsTesting](https://github.com/pointfreeco/xctest-dynamic-overlay/blob/5ede785843718dc6619b57b22a735bed541de7c9/Sources/XCTestDynamicOverlay/XCTIsTesting.swift#L4) over something like `let isTesting = NSClassFromString("XCTestCase") != nil`.

Just for the sake of understanding, I'm wondering how the contents of `_XCTIsTesting` was created, and what situations it covers that a simpler check like `NSClassFromString("XCTestCase") != nil` wouldn't account for.

**Comment by mbrandonw** [comment, marked as answer]
Hi @dafurman, `NSClassFromString` is not available on all the platforms we support. If you are just developing on Apple platforms then the `NSClassFromString` is probably ok, as far as I know.
